spot-vaults/contracts/_utils/UniswapV3PoolHelpers.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// solhint-disable-next-line compiler-version
+pragma solidity ^0.7.6;
+
+import { FullMath } from "@uniswap/v3-core/contracts/libraries/FullMath.sol";
+import { TickMath } from "@uniswap/v3-core/contracts/libraries/TickMath.sol";
+import { IUniswapV3Pool } from "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+
+/**
+ *  @title UniswapV3PoolHelpers
+ *
+ *  @notice Library with helper functions for a UniswapV3Pool.
+ *
+ */
+library UniswapV3PoolHelpers {
+    /// @notice Calculates the Time-Weighted Average Price (TWAP) given the TWAP tick and unit token amounts.
+    /// @param twapTick The Time-Weighted Average Price tick.
+    /// @param token0UnitAmt The fixed-point amount of token0 equivalent to 1.0.
+    /// @param token1UnitAmt The fixed-point amount of token1 equivalent to 1.0.
+    /// @param one 1.0 represented in the same fixed point denomination as calculated TWAP.
+    /// @return The computed TWAP price.
+    function calculateTwap(
```

Comment:
All this comes from existing code
https://github.com/ampleforth/spot/blob/d6e9cf53507b4d76f769fe6b651330c4e055bd3e/spot-vaults/contracts/WethWamplManager.sol#L300